common/motion_utils/include/motion_utils/trajectory/trajectory.hpp

```diff
@@ -437,6 +437,39 @@ double calcSignedArcLength(const T & points, const size_t src_idx, const size_t
   return dist_sum;
 }
 
+/**
+ *  @brief Computes the partial sums of the elements in the sub-ranges of
+ *         the range [src_idx, dst_idx) and return these sum as vector
+ */
+template <class T>
+std::vector<double> calcSignedArcLengthPartialSum(
```

Comment:
motion_utils also have a way to calculate a partial sum. Is this function different from that?
https://github.com/autowarefoundation/autoware.universe/blob/main/common/motion_utils/include/motion_utils/trajectory/trajectory.hpp/#L420

Reply:
yes, that is true, however, [this function](https://github.com/autowarefoundation/autoware.universe/blob/main/common/motion_utils/include/motion_utils/trajectory/trajectory.hpp/#L420) returns the current partial sum, instead of std::vector<double> of partial sums. 

What happen is that because [here](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/src/path_utilities.cpp#:~:text=int64_t%3E%20unique_lane_ids%3B-,for%20(size_t%20i%20%3D%200%3B%20i%20%3C%20path.points.,const%20double%20s%20%3D%20motion_utils%3A%3AcalcSignedArcLength(transformed_path%2C%200%2C%20i)%3B,-for%20(const), the `calcSignedArcLength` is recalled, the loop will re-run again. With vector of partial sums, we only need to compute the partial sum once.

Reply:
thanks!